docs/the-new-architecture/backward-compatibility-fabric-components.md

```diff
@@ -0,0 +1,432 @@
+
+## Update the codebase
+
+### iOS
+
+The second step is to instruct Xcode to avoid compiling all the lines using the New Architecture types and files when we are building an app with the Old Architecture.
+
+A Fabric Component requires an header file and an implementation file to add the actual `View` to the module.
+
+For example, the `RNMyComponentView.h` header file could look like this:
+
+```objective-c
+#import <React/RCTViewComponentView.h>
+#import <UIKit/UIKit.h>
+
+#ifndef NativeComponentExampleComponentView_h
+#define NativeComponentExampleComponentView_h
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RNMyComponentView : RCTViewComponentView
+@end
+
+NS_ASSUME_NONNULL_END
+
+#endif /* NativeComponentExampleComponentView_h */
+```
+
+The implementation `RNMyComponentView.mm` file, instead, could look like this:
+
+```objective-c
+#import "RNMyComponentView.h"
+
+// <react/renderer imports>
+
+#import "RCTFabricComponentsPlugins.h"
+
+using namespace facebook::react;
+
+@interface RNMyComponentView () <RCTMyComponentViewViewProtocol>
+
+@end
+
+@implementation RNMyComponentView {
+    UIView * _view;
+}
+
++ (ComponentDescriptorProvider)componentDescriptorProvider
+{
+    // ... return the descriptor ...
+}
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+  // ... initialize the object ...
+}
+
+- (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
+{
+  // ... set up the props ...
+
+  [super updateProps:props oldProps:oldProps];
+}
+
+Class<RCTComponentViewProtocol> MyComponentViewCls(void)
+{
+  return RNMyComponentView.class;
+}
+
+@end
+```
+
+To make sure that Xcode skips these files, we can wrap **both** of them in some `#ifdef RCT_NEW_ARCH_ENABLED` compilation pragma. For example, the header file could change as follows:
+
+```diff
++ #ifdef RCT_NEW_ARCH_ENABLED
+#import <React/RCTViewComponentView.h>
+#import <UIKit/UIKit.h>
+
+// ... rest of the header file ...
+
+#endif /* NativeComponentExampleComponentView_h */
++ #endif
+```
+
+The same two lines should be added in the implementation file, as first and last lines.
+
+The above snippet uses the same `RCT_NEW_ARCH_ENABLED` flag used in the previous [section](#dependencies-ios). When this flag is not set, Xcode skips the lines within the `#ifdef` during compilation and it does not include them into the compiled binary. The compiled binary will have a the `RNMyComponentView.o` object but it will be an empty object.
+
+### Android
+
+As we can't use conditional compilation blocks on Android, we will define two different source sets. This will allow to create a backward compatible TurboModule with the proper source that is loaded and compiled depending on the used architecture.
+
+Therefore, you have to:
+
+1. Create a Native Component in the `src/oldarch` path. See [this guide](../native-components-intro) to learn how to create a Native Component.
```

Comment:
```suggestion
1. Create a Native Component in the `src/oldarch` path. See [this guide](../native-components-intro) to learn how to create a Native Component.
```

This link doesn't work form me locally, and it's also reported by Docusaurus, however I'm not sure which page you want to reference here, there is no doc using `native-components-intro` id.